src/html_builder.ts

```diff
@@ -15,486 +15,499 @@ import {
-
-            let wdt_back = 1.0 * (outer_rad - inner_rad);
-            let wdt_frnt = srt_prop * (outer_rad - inner_rad);
-
-            let srk_back = 2 * Math.PI * rad_back * srk_prop;
-            let gap_back = 2 * Math.PI * rad_back * gap_prop;
-            let off_back = 2 * Math.PI * rad_back * cum_prop;
-
-            let srk_frnt = 2 * Math.PI * rad_frnt * srk_prop;
-            let gap_frnt = 2 * Math.PI * rad_frnt * gap_prop;
-            let off_frnt = 2 * Math.PI * rad_frnt * cum_prop;
-
-            let ths_id = class_ids[i];
-            let ths_col = class_defs[i]["color"];
-            // TODO should names also go on the id dialog?
-            // let ths_nam = class_defs[i]["name"];
-            dialog_html += `
-            <circle
-                r="${rad_back}" cx="${center_coord}" cy="${center_coord}" 
-                stroke="${ths_col}" 
-                fill-opacity="0"
-                stroke-opacity="${cl_opacity}"
-                stroke-width="${wdt_back}"; 
-                stroke-dasharray="${srk_back} ${gap_back}" 
-                stroke-dashoffset="${off_back}" />
-            <circle
-                id="${idd_id}__circ_${ths_id}"
-                r="${rad_frnt}" cx="${center_coord}" cy="${center_coord}"
-                fill-opacity="0"
-                stroke="${ths_col}" 
-                stroke-opacity="1.0"
-                stroke-width="${wdt_frnt}" 
-                stroke-dasharray="${srk_frnt} ${gap_frnt}" 
-                stroke-dashoffset="${off_frnt}" />
-            `;
+            let this_id: string = class_ids[i].toString();
+            let this_color: string = ulabel.color_info[this_id]
+            // let this_color: string = ulabel.subtasks[st]["class_defs"][i]["color"];
+            let this_name: string = ulabel.subtasks[st]["class_defs"][i]["name"];
+
+            let sel: string = "";
+            let href: string = ' href="#"';
+            if (i == 0) {
+                sel = " sel";
+                href = "";
+            }
+            if (ulabel.config["allow_soft_id"]) {
+                let msg = "Only hard id is currently supported";
+                throw new Error(msg);
+            }
+            else {
+                toolbox_html += `
+                    <a${href} id="${tbid}_sel_${this_id}" class="tbid-opt${sel}">
+                        <div class="colprev ${tbid}_colprev_${this_id}" style="background-color: ${this_color}"></div> <span class="tb-cls-nam">${this_name}</span>
+                    </a>
+                `;
+            }
         }
-        dialog_html += `
-            </svg>
-            <div class="centcirc"></div>
+        toolbox_html += `
         </div>`;
 
-        return dialog_html;
+        // Add dialog to the document
+        front_subtask_dialog_container_jq.append(front_dialog_html_v2); // TODO(new3d) MOVE THIS TO GLOB BOX -- superimpose atop thee anchor already there when needed, no remove and add back
```

Comment:
Is this comment still relavent?
```suggestion
        front_subtask_dialog_container_jq.append(front_dialog_html_v2);
```

Reply:
No, good catch